### PR TITLE
[Doc] stream load supports compression during transmission

### PR DESCRIPTION
**Pull request by evelynzhaojie**
Signed-off-by: evelynzhaojie <everlyn.zhaojie@gmail.com

## Why I'm doing:

https://github.com/StarRocks/starrocks/pull/43732 new feature

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md [hunk 1]
```diff
@@ -121,6 +125,7 @@ The parameters in the `data_desc` descriptor can be divided into three types: co
...
 | ignore_json_size  | No       | Specifies whether to check the size of the JSON body in the HTTP request.<br/>**NOTE**<br/>By default, the size of the JSON body in an HTTP request cannot exceed 100 MB. If the JSON body exceeds 100 MB in size, an error "The size of this batch exceed the max size [104857600] of json type data data [8617627793]. Set ignore_json_size to skip check, although it may lead huge memory consuming." is reported. To prevent this error, you can add `"ignore_json_size:true"` in the HTTP request header to instruct StarRocks not to check the JSON body size. |
+| compression, Content-Encoding | NO | The encoding algorithm that is applied to the data during transmission. Supported algorithms include GZIP, BZIP2, LZ4_FRAME, DEFLATE, and ZSTD。|
```
Comment (rickif) [inline docs/en/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md]: DEFLATE is not ready yet. Please remove it.

File: docs/zh/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md [hunk 2]
```diff
@@ -123,8 +127,9 @@ http://<fe_host>:<fe_http_port>/api/<database_name>/<table_name>/_stream_load
...
 | json_root         | 否           | 用于指定待导入 JSON 数据的根元素。仅在使用匹配模式导入 JSON 数据时需要指定该参数。参数取值为合法的 JsonPath 字符串。默认值为空，表示会导入整个 JSON 数据文件的数据。具体请参见本文提供的示例“[导入数据并指定 JSON 根节点](#指定-json-根节点使用匹配模式导入数据)”。 |
 | ignore_json_size | 否   | 用于指定是否检查 HTTP 请求中 JSON Body 的大小。<br />**说明**<br />HTTP 请求中 JSON Body 的大小默认不能超过 100 MB。如果 JSON Body 的大小超过 100 MB，会提示 "The size of this batch exceed the max size [104857600] of json type data data [8617627793]. Set ignore_json_size to skip check, although it may lead huge memory consuming." 错误。为避免该报错，可以在 HTTP 请求头中添加 `"ignore_json_size:true"` 设置，忽略对 JSON Body 大小的检查。 |
+| compression, Content-Encoding | 否 | 指定在 STREAM LOAD 数据传输过程中使用哪种压缩算法，支持 GZIP、BZIP2、LZ4_FRAME、ZSTD 算法。示例：`curl --location-trusted -u root:  -v 'http://127.0.0.1:18030/api/db0/tbl_simple/_stream_load' \-X PUT  -H "expect:100-continue" \-H 'format: json' -H 'compression: lz4_frame'   -T ./b.json.lz4`。 |
 
-另外，导入 JSON 格式的数据时，需要注意单个 JSON 对象的大小不能超过 4 GB。如果 JSON 文件中单个 JSON 对象的大小超过 4 GB，会提示 "This parser can't support a document that big." 错误。
+另外，导入 JSON 格式的数据时，需要注意单个 JSON 对象的大小不能超过 4 GB。如果 |JSON 文件中单个 JSON 对象的大小超过 4 GB，会提示 "This parser can't support a document that big." 错误。
```
Comment (jaogoy) [inline docs/zh/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md:132]: JSON前多个 |

File: docs/en/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md (hunk 1 above)
Comment (evelynzhaojie) [reply docs/en/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md]: done

File: docs/zh/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md (hunk 2 above)
Comment (evelynzhaojie) [reply docs/zh/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md]: done

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Pull request merged**